x/asset/keeper/msg_server_authorize_address.go

```diff
@@ -10,23 +10,25 @@ import (
 	"github.com/realiotech/realio-network/x/asset/types"
 )
 
-func (k msgServer) AuthorizeAddress(goCtx context.Context, msg *types.MsgAuthorizeAddress) (*types.MsgAuthorizeAddressResponse, error) {
-	ctx := sdk.UnwrapSDKContext(goCtx)
-
+func (ms msgServer) AuthorizeAddress(goCtx context.Context, msg *types.MsgAuthorizeAddress) (*types.MsgAuthorizeAddressResponse, error) {
 	// Check if the value exists
-	token, isFound := k.GetToken(ctx, msg.Symbol)
-	if !isFound {
-		return nil, errorsmod.Wrapf(sdkerrors.ErrKeyNotFound, "symbol %s does not exists", msg.Symbol)
+	token, err := ms.Token.Get(goCtx, types.TokenKey(msg.Symbol))
+	if err != nil {
+		return nil, errorsmod.Wrapf(sdkerrors.ErrKeyNotFound, "symbol %s does not exists : %s", msg.Symbol, err.Error())
 	}
 
 	// Checks if the token manager signed
-	signers := msg.GetSigners()
+	signers, _, err := ms.cdc.GetMsgV1Signers(msg)
+	if err != nil {
+		return nil, err
+	}
+
 	if len(signers) != 1 {
 		return nil, errorsmod.Wrap(sdkerrors.ErrUnauthorized, "invalid signers")
 	}
 
 	// assert that the manager account is the only signer of the message
-	if signers[0].String() != token.Manager {
+	if sdk.AccAddress(signers[0]).String() != token.Manager {
```

Comment:
Can we just compare `msg.Manager` with `token.Manager` and validate signer in function GetSigner